### ISSUE TITLE
shellcheck treats "typeset" and "declare" differently

### ISSUE THREAD

**Issue by marcone** [koalaman/shellcheck#1922]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

function testit {
  if typeset -F foobar
  then
    foobar "$@"
  else
    echo "$@"
  fi
}

```

#### Here's what shellcheck currently says:
```
Line 4	SC2034: foobar appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

Since `typeset` and `declare` are really the same command/builtin, shellcheck should not treat them differently. In this case I think the behavior that shellcheck exhibits for `declare` is correct.

**Comment by masalomon** [comment]
I came across this while reporting an unnecessary warning with `typeset -x`, which is not given for its synonym `declare -x`.  I suspect the problem can be resolved by adding code for `typeset` to the function `getReferencedVariableCommand` in `ShellCheck/AnalyzerLib.hs` (c. line 545 in the current master).

**Comment by e-kwsm** [comment]
Seems to be fixed (v0.11.0).

**Comment by masalomon** [comment]
Fixed in issue #2354, in commit 788aee1b7c1c74863c85837fbd53dd45cf810d14.
